pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -241,7 +241,9 @@ func (a *Bootstrap) addStorageFiles(base string, uri string, templateData *boots
 		mode = 0600
 	}
 	ign := ignition.FileFromBytes(strings.TrimSuffix(base, ".template"), "root", mode, data)
-	ign.Append = appendToFile
+	if appendToFile {
+		ign = ignition.FileAppendFromBytes(strings.TrimSuffix(base, ".template"), "root", mode, data)
```

Comment:
Instead of setting up `ign` and then, in the append case, immediately clobbering it, I'd rather have:

```go
var ign whateverTypeThisShouldBe
if appendToFile {
  ign = ignition.FileAppendFromBytes(strings.TrimSuffix(base, ".template"), "root", mode, data)
} else {
  ign = ignition.FileFromBytes(strings.TrimSuffix(base, ".template"), "root", mode, data)
}
```

Or we should update `FileFromBytes` to support an `append` parameter, or an `options` struct parameter, or some such...

Reply:
yeah, I've already accounted for that, just want to validate that the v3 way of doing append is how I did @ajeddeloh

Reply:
yeah this looks good.

Reply:
Actually, in the case of appending you'll want to make sure a file entry doesn't already exist at that path. In that case you'll need to add the `append` entry to it's `append` list.

Reply:
> Actually, in the case of appending you'll want to make sure a file entry doesn't already exist at that path.

This helper is for our internal use, so we can probably ignore that case.  Will Ignition crash and burn if this happens, so we can fail out in CI if by some programmer mistake we have two entries for the same file?

Reply:
Ignition will crash and burn at validation time (note: you can call the validation function ahead of time) since there will be two file entries with the same `path`